python/cudf/cudf/core/column/categorical.py

```diff
@@ -949,7 +949,7 @@ def __cuda_array_interface__(self):
     def to_pandas(self, index=None):
         signed_dtype = min_signed_type(len(self.categories))
         codes = self.cat().codes.astype(signed_dtype).fillna(-1).to_array()
-        categories = self.categories.to_pandas()
+        categories = self.categories.to_arrow().to_pandas()
```

Comment:
Can use `self.categories.values_host` instead of using `to_arrow().to_pandas()`

Reply:
There is a subtle issue with this approach unfortunately. `values_host` is just an alias for `to_pandas().values`, but now `to_pandas()` returns nullable dtypes instead of numpy types, and right now you can't use a pandas nullable dtype as `categories` in pandas, you'll just get `object` instead. For example:

```
>>> codes = [0,1,2]
>>> categories = pd.Series([100,200,300], dtype=pd.Int64Dtype())
>>> pd_cat_result = pd.Categorical.from_codes(codes, categories=categories)
>>> pd_cat_result
[100, 200, 300]
Categories (3, object): [100, 200, 300]
>>> pd_cat_result.categories.dtype
dtype('O')
```
This is the reason I changed this - we need to bunny-hop our own `to_pandas()` function in this case because it doesn't result in the right categorical dtype on the pandas side.